core/bin/external_node/src/main.rs

```diff
@@ -337,14 +340,18 @@ async fn run_core(
     )
     .await?;
 
-    let l1_batch_commit_data_generator: Arc<dyn L1BatchCommitDataGenerator> = match config
-        .optional
-        .l1_batch_commit_data_generator_mode
-    {
-        L1BatchCommitDataGeneratorMode::Rollup => Arc::new(RollupModeL1BatchCommitDataGenerator {}),
-        L1BatchCommitDataGeneratorMode::Validium => {
-            Arc::new(ValidiumModeL1BatchCommitDataGenerator {})
-        }
+    let (l1_batch_commit_data_generator, input_generator): (
+        Arc<dyn L1BatchCommitDataGenerator>,
+        Box<dyn InputGenerator>,
+    ) = match config.optional.l1_batch_commit_data_generator_mode {
+        L1BatchCommitDataGeneratorMode::Rollup => (
+            Arc::new(RollupModeL1BatchCommitDataGenerator {}),
+            Box::new(RollupInputGenerator),
+        ),
+        L1BatchCommitDataGeneratorMode::Validium => (
+            Arc::new(ValidiumModeL1BatchCommitDataGenerator {}),
+            Box::new(ValidiumInputGenerator),
+        ),
     };
```

Comment:
Side comment: I'm starting to feel that we need to revisit all the differences we have between rollup and validium, and probably have a single abstraction that is capable of handling all the differences (right now I'm not sure if that's possible, just thinking out loud).

The current situation is still strictly better than having `if`s here and there, since we instantiate all the stuff in a single place and the underlying components don't have to think about it, but still it somewhat bugs me that we in theory can have rollup commit data generator with validium input generator.

No action required here.

Reply:
i agree, i thought about forking the commitment_generator folder for validiums but i didnt want to redo what we have for multivms (introduces bloat) and the even within validiums the difference will likely only change the single `pubdataCommitments` field on `CommitBatchInfo`.

If we changed the commitment_generator to be the base of the data and then the input generator took in the raw data and computed the actual input then maybe that seems like a better pipeline. wdyt?

Reply:
I think it's fine to keep it as is for now and revisit a bit later.